src/session.ts

```diff
@@ -0,0 +1,133 @@
+import * as superagent from 'superagent';
+
+import debugModule from 'debug';
+const debug = debugModule('everyauth:profile');
+
+import { getAuthedProfile, IProfile } from './profile';
+import EveryAuthVersion from './version';
+import { getInstallIdByTags } from './install';
+
+import { SERVICE_TAG, USER_TAG, TENANT_TAG } from './constants';
+
+const COMMIT_URL_SUFFIX = '/commit';
+
+const META_INTEGRATION_ID = 'everyauth';
+
+export interface ISession {
+  id: string;
+  userId: string;
+  tenantId?: string;
+
+  tags: Record<string, string | null>;
+}
+
+const getSessionUrl = (profile: IProfile) =>
+  `${profile.baseUrl}/v2/account/${profile.account}/subscription/${profile.subscription}/integration/${META_INTEGRATION_ID}`;
+
+export const start = async (
+  serviceId: string,
+  tenantId: string | undefined,
+  userId: string,
+  hostedBaseUrl: string
+): Promise<string> => {
+  const profile = await getAuthedProfile();
+  const baseUrl = getSessionUrl(profile);
+
+  const payload: { redirectUrl: string; tags: Record<string, string>; components: string[]; installId?: string } = {
+    redirectUrl: `${hostedBaseUrl}${COMMIT_URL_SUFFIX}`,
+    tags: {
+      [SERVICE_TAG]: serviceId,
+      [USER_TAG]: userId,
+      ...(tenantId
+        ? {
+            [TENANT_TAG]: tenantId,
+          }
+        : {}),
+    },
+    components: [serviceId],
+  };
+
+  // Is there an existing matching user?
+  try {
+    payload.installId = await getInstallIdByTags({ serviceId, userId, tenantId });
+    debug(`${tenantId || ''}${tenantId ? '/' : ''}${userId}: found matching install, reusing ${payload.installId}`);
+  } catch (e) {
```

Comment:
That is what I was referring to in my comment in `getInstallIdByTags` - absence of a matching install is a "normal" application state and the dev should not need to detect it by catching exception.

Reply:
This is an internal endpoint, but still - adjusted to return undefined on not found.